software/contrib/harmonic_lfos.py

```diff
@@ -95,26 +160,43 @@ def display_selected_lfo(self):
         
         number = self.divisions[self.selected_lfo]
         x = 2 if number >= 10 else 6
-        oled.text(str(number),x,20,1)
+        oled.text(str(number),x,12,1)
+        
+        lfo_mode = self.modes[self.selected_lfo]
+        if lfo_mode == 0:
+            self.draw_wave('sine')
+        elif lfo_mode == 1:
+            self.draw_wave('saw')
+        elif lfo_mode == 2:
+            self.draw_wave('square')
+        elif lfo_mode == 3:
+            self.draw_wave('off')
+        elif lfo_mode == 4:
+            self.draw_wave('random')
```

Comment:
Optional: There's no need to pass the string value of the selected LFO since its already an instance variable. You can simplify this block of code down to just 
```python
self.draw_wave()
```
And then update the `draw_wave()` method to:
```python
def draw_wave(self):
    if self.selected_lfo == 0:  # square
        ...
    elif self.selected_lfo == 1:  # saw
        ...
```